### PR TITLE
Bugfix/1176 error formatting generated code

### LINKED ISSUE #1176: Javac assertion error during generation with CLI
During generation I am getting this exception. Please note that I am not generating any Java output (only HTML and TypeScript). I am clueless what this error should tell me and where the actual problem could lie:
```
Exception in thread "main" java.lang.AssertionError
	at org.openjdk.tools.javac.util.Assert.error(Assert.java:155)
	at org.openjdk.tools.javac.util.Assert.check(Assert.java:46)
	at org.openjdk.tools.javac.parser.JavacParser.reportSyntaxError(JavacParser.java:481)
	at org.openjdk.tools.javac.parser.JavacParser.reportSyntaxError(JavacParser.java:462)
	at org.openjdk.tools.javac.parser.JavacParser.accept(JavacParser.java:511)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3555)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBodyDeclaration(JavacParser.java:3599)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceBody(JavacParser.java:3549)
	at org.openjdk.tools.javac.parser.JavacParser.classDeclaration(JavacParser.java:3398)
	at org.openjdk.tools.javac.parser.JavacParser.classOrInterfaceOrEnumDeclaration(JavacParser.java:3352)
	at org.openjdk.tools.javac.parser.JavacParser.typeDeclaration(JavacParser.java:3341)
	at org.openjdk.tools.javac.parser.JavacParser.parseCompilationUnit(JavacParser.java:3182)
	at com.google.googlejavaformat.java.Formatter.format(Formatter.java:144)
	at com.google.googlejavaformat.java.Formatter.getFormatReplacements(Formatter.java:258)
	at com.google.googlejavaformat.java.Formatter.formatSource(Formatter.java:234)
	at com.google.googlejavaformat.java.Formatter.formatSource(Formatter.java:202)
	at com.devonfw.cobigen.cli.utils.ParsingUtils.formatJavaSources(ParsingUtils.java:244)
	at com.devonfw.cobigen.cli.commands.GenerateCommand.generate(GenerateCommand.java:350)
	at com.devonfw.cobigen.cli.commands.GenerateCommand.call(GenerateCommand.java:155)
	at com.devonfw.cobigen.cli.commands.GenerateCommand.call(GenerateCommand.java:46)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1680)
	at picocli.CommandLine.access$900(CommandLine.java:144)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2038)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2005)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1872)
	at picocli.CommandLine.execute(CommandLine.java:1801)
	at com.devonfw.cobigen.cli.CobiGenCLI.main(CobiGenCLI.java:43)
```

### PR DESCRIPTION
**Pull request by lilliCao**
Fixes #1176 

Implements

* Add check if the generated files are java files or not before formatting

@devonfw/cobigen

### CODE REVIEW

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java [hunk 1]
```diff
@@ -345,13 +348,18 @@ public void generate(File inputFile, File inputProject, List<? extends Generable
...
-                try {
-                    ParsingUtils.formatJavaSources(report.getGeneratedFiles());
-                } catch (FormatterException e) {
-                    LOG.info(
-                        "Generation was successful but we were not able to format your code. Maybe you will see strange formatting.",
-                        LOG.isDebugEnabled() ? e : null);
+            if (ValidationUtils.checkGenerationReport(report)) {
+                Set<Path> generatedJavaFiles = report.getGeneratedFiles().stream()
+                    .filter(e -> FileUtils.getExtension(e.toAbsolutePath().toString()).equals("java"))
+                    .collect(Collectors.toSet());
```
Comment (maybeec) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:354]: As of performance, I would like to go with this approach: https://stackoverflow.com/a/25188854
You could even already reuse the Pattern in the above changes where already for the extension is checked for java / yml / yaml.

Comment (lilliCao) [reply cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:354]: I could absolutely implement this. But does pattern/matcher really improve the performance? The string compare is computational expensive?

Comment (maybeec) [reply cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:354]: I think the compare is not an issue, but constructing the absolutepath or actually the path string of a long path is more expensive.

You could even think of asking path.getLastSegment().endsWith(".java"), which would even be very performant. Maybe even more.

Comment (lilliCao) [reply cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:354]: If it is not about the comparing, then either with pattern or contains/equals I do need to construct the absolute path. Unfortunately we are using java.nio.Path which does not have the getLastSegment() method

**Pull request merged**